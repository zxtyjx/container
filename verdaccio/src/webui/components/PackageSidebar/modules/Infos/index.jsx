import React from 'react';
import PropTypes from 'prop-types';
import Module from '../../Module';
import ModuleContentPlaceholder from '../../ModuleContentPlaceholder';

import classes from './style.scss';

const renderSection = (title, url) => (
  <li>
    <span>{title}</span>
    <a href={url} target="_blank">
      {url}
    </a>
  </li>
);

const Infos = ({homepage, repository, license}) => {
  const showInfo = homepage || repository || license;
  return <Module title="Infos" className={classes.infosModule}>
      {showInfo ? <ul>
          {homepage && renderSection('Homepage', homepage)}
          {repository && renderSection('Repository', repository)}
          {license && <li>
              <span>License</span>
              <span>{license}</span>
            </li>}
        </ul> : <ModuleContentPlaceholder text="Not Available!" />}
    </Module>;
};

Infos.propTypes = {
  homepage: PropTypes.string,
  repository: PropTypes.string,
  license: PropTypes.string
};

export default Infos;
