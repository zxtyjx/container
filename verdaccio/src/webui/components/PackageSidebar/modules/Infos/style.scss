@import '../../../../styles/variable';

.infosModule {
  li {
    display: flex;
    font-size: 14px;
    line-height: 2;

    a {
      color: inherit;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
      max-width: 150px;
    }
    
    a:last-child,
    span:last-child {
      margin-left: auto;
    }
  }
}
