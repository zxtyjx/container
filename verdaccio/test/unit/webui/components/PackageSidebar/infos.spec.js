/**
 * Infos component
 */

import React from 'react';
import { shallow } from 'enzyme';
import Infos from '../../../../../src/webui/components/PackageSidebar/modules/Infos/index';

describe('<PackageSidebar /> : <Infos />', () => {
  it('should load the component without props', () => {
    const wrapper = shallow(<Infos/>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should load the Info component with props', () => {
    const props = {
      homepage: 'https://www.verdaccio.org',
      license: 'MIT',
      repository: 'https://github.com/verdaccio/verdaccio'
    }
    const wrapper = shallow(<Infos {...props}/>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should load the Info component with homepage only', () => {
    const props = {
      homepage: 'https://www.verdaccio.org'
    }
    const wrapper = shallow(<Infos {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should load the Info component with license only', () => {
    const props = {
      license: 'MIT',
    }
    const wrapper = shallow(<Infos {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should load the Info component with repository only', () => {
    const props = { repository: 'https://github.com/verdaccio/verdaccio' };
    const wrapper = shallow(<Infos {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
