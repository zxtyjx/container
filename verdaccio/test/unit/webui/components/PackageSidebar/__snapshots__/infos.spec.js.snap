// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageSidebar /> : <Infos /> should load the Info component with homepage only 1`] = `"<div class=\\"module infosModule\\"><h2 class=\\"moduleTitle\\">Infos</h2><div><ul><li><span>Homepage</span><a href=\\"https://www.verdaccio.org\\" target=\\"_blank\\">https://www.verdaccio.org</a></li></ul></div></div>"`;

exports[`<PackageSidebar /> : <Infos /> should load the Info component with license only 1`] = `"<div class=\\"module infosModule\\"><h2 class=\\"moduleTitle\\">Infos</h2><div><ul><li><span>License</span><span>MIT</span></li></ul></div></div>"`;

exports[`<PackageSidebar /> : <Infos /> should load the Info component with props 1`] = `"<div class=\\"module infosModule\\"><h2 class=\\"moduleTitle\\">Infos</h2><div><ul><li><span>Homepage</span><a href=\\"https://www.verdaccio.org\\" target=\\"_blank\\">https://www.verdaccio.org</a></li><li><span>Repository</span><a href=\\"https://github.com/verdaccio/verdaccio\\" target=\\"_blank\\">https://github.com/verdaccio/verdaccio</a></li><li><span>License</span><span>MIT</span></li></ul></div></div>"`;

exports[`<PackageSidebar /> : <Infos /> should load the Info component with repository only 1`] = `"<div class=\\"module infosModule\\"><h2 class=\\"moduleTitle\\">Infos</h2><div><ul><li><span>Repository</span><a href=\\"https://github.com/verdaccio/verdaccio\\" target=\\"_blank\\">https://github.com/verdaccio/verdaccio</a></li></ul></div></div>"`;

exports[`<PackageSidebar /> : <Infos /> should load the component without props 1`] = `"<div class=\\"module infosModule\\"><h2 class=\\"moduleTitle\\">Infos</h2><div><p class=\\"emptyPlaceholder\\">Not Available!</p></div></div>"`;
